core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1322,6 +1327,30 @@ class KafkaApis(val requestChannel: RequestChannel,
     txnCoordinator.handleInitPid(initPidRequest.transactionalId, sendResponseCallback)
   }
 
+  def handleEndTransactionRequest(request: Request): Unit = {
+    throw new UnsupportedOperationException
```

Comment:
It would be good to add a "Not implemented yet" message to all of these exceptions.